lib/driverProviders/driverProvider.ts

```diff
@@ -93,56 +92,48 @@ export abstract class DriverProvider {
    *
    * @param drivers {webdriver.WebDriver[]} The webdriver instances
    */
-  static quitDrivers(provider: DriverProvider, drivers: WebDriver[]): q.Promise<void> {
-    let deferred = q.defer<void>();
-    wdpromise
-        .all(drivers.map((driver: WebDriver) => {
-          return provider.quitDriver(driver);
+  static async quitDrivers(provider: DriverProvider, drivers: WebDriver[]): Promise<void> {
+    return await Promise
+        .all(drivers.map(async (driver: WebDriver) => {
+          await provider.quitDriver(driver);
         }))
-        .then(
-            () => {
-              deferred.resolve();
-            },
-            () => {
-              deferred.resolve();
-            });
-    return deferred.promise;
+        .then(() => {});
   }
 
   /**
    * Default update job method.
    * @return a promise
    */
-  updateJob(update: any): q.Promise<any> {
-    return q.fcall(function() {});
+  updateJob(update: any): Promise<any> {
+    return Promise.resolve(() => {});
   };
 
   /**
    * Default setup environment method, common to all driver providers.
    */
-  setupEnv(): q.Promise<any> {
+  async setupEnv(): Promise<any> {
     let driverPromise = this.setupDriverEnv();
     if (this.config_.useBlockingProxy && !this.config_.blockingProxyUrl) {
       // TODO(heathkit): If set, pass the webDriverProxy to BP.
-      return driverPromise.then(() => this.bpRunner.start());
+      await driverPromise.then(() => this.bpRunner.start());
```

Comment:
No need to chain here. Actually, this whole thing could be a lot simpler

```
await this.setupDriverEnv();
if (this.config_.useBlockingProxy && !this.config_.blockingProxyUrl) {
   await this.bpRunner.start()
}
```

Reply:
Fixed.